tests/std/tests/P0202R3_constexpr_algorithm_and_exchange/test.cpp

```diff
@@ -488,113 +464,114 @@ constexpr bool test_pop_heap_and_push_heap() {
     };
     assert(is_heap(begin(expectedPushed), end(expectedPushed)));
     assert(equal(begin(buff), end(buff), begin(expectedPushed), end(expectedPushed)));
-    return true;
 }
 
-constexpr bool test_permutations() {
-    int buff[] = {1, 2, 3, 4};
-    // using std::array here is TRANSITION, DevCom-892153
-    constexpr array<const int, 4> expected[] = {
-        array<const int, 4>{10, 20, 30, 40},
-        array<const int, 4>{10, 20, 40, 30},
-        array<const int, 4>{10, 30, 20, 40},
-        array<const int, 4>{10, 30, 40, 20},
-        array<const int, 4>{10, 40, 20, 30},
-        array<const int, 4>{10, 40, 30, 20},
-        array<const int, 4>{20, 10, 30, 40},
-        array<const int, 4>{20, 10, 40, 30},
-        array<const int, 4>{20, 30, 10, 40},
-        array<const int, 4>{20, 30, 40, 10},
-        array<const int, 4>{20, 40, 10, 30},
-        array<const int, 4>{20, 40, 30, 10},
-        array<const int, 4>{30, 10, 20, 40},
-        array<const int, 4>{30, 10, 40, 20},
-        array<const int, 4>{30, 20, 10, 40},
-        array<const int, 4>{30, 20, 40, 10},
-        array<const int, 4>{30, 40, 10, 20},
-        array<const int, 4>{30, 40, 20, 10},
-        array<const int, 4>{40, 10, 20, 30},
-        array<const int, 4>{40, 10, 30, 20},
-        array<const int, 4>{40, 20, 10, 30},
-        array<const int, 4>{40, 20, 30, 10},
-        array<const int, 4>{40, 30, 10, 20},
-        array<const int, 4>{40, 30, 20, 10},
+constexpr void test_permutations() {
+    int buff[] = {10, 20, 30, 40};
+
+    constexpr int expected[24][4] = {
+        {10, 20, 30, 40},
+        {10, 20, 40, 30},
+        {10, 30, 20, 40},
+        {10, 30, 40, 20},
+        {10, 40, 20, 30},
+        {10, 40, 30, 20},
+        {20, 10, 30, 40},
+        {20, 10, 40, 30},
+        {20, 30, 10, 40},
+        {20, 30, 40, 10},
+        {20, 40, 10, 30},
+        {20, 40, 30, 10},
+        {30, 10, 20, 40},
+        {30, 10, 40, 20},
+        {30, 20, 10, 40},
+        {30, 20, 40, 10},
+        {30, 40, 10, 20},
+        {30, 40, 20, 10},
+        {40, 10, 20, 30},
+        {40, 10, 30, 20},
+        {40, 20, 10, 30},
+        {40, 20, 30, 10},
+        {40, 30, 10, 20},
+        {40, 30, 20, 10},
     };
 
-    auto cursor = begin(expected);
+    size_t cursor = 0;
     do {
-        assert(equal(begin(buff), end(buff), cursor->begin(), cursor->end()));
+        assert(equal(begin(buff), end(buff), begin(expected[cursor]), end(expected[cursor])));
         ++cursor;
     } while (next_permutation(begin(buff), end(buff)));
 
-    assert(cursor == end(expected));
+    assert(cursor == 24);
+    assert(equal(begin(buff), end(buff), begin(expected[0]), end(expected[0])));
+
     assert(!prev_permutation(begin(buff), end(buff)));
+    cursor = 23;

```

Comment:
I think it would be cleaner if the cursor decrement on line 513 was moved above the assert on line 512 instead (assert on line 516 would just compare to 0)